subcommand/cert-manager/command.go

```diff
@@ -0,0 +1,229 @@
+	"github.com/hashicorp/consul-k8s/subcommand/flags"
+	"github.com/mitchellh/cli"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+type Command struct {
+	flagSet   *flag.FlagSet
+	httpFlags *flags.HTTPFlags
+	UI        cli.Ui
+
+	flagAutoName        string // MutatingWebhookConfiguration for updating
+	flagAutoHosts       string // SANs for the auto-generated TLS cert.
+	flagSecretName      string // Name of secret where certificates will be written to.
+	flagSecretNamespace string // Namespace of the secret where certificates will be written to.
+	flagCertFile        string // TLS cert for listening (PEM)
+	flagKeyFile         string // TLS cert private key (PEM)
+
+	clientset kubernetes.Interface
+
+	once sync.Once
+	help string
+}
+
+var (
+	certGenLog = ctrl.Log.WithName("webhook-cert-gen")
+)
+
+func (c *Command) init() {
+	c.flagSet = flag.NewFlagSet("", flag.ContinueOnError)
+	c.flagSet.StringVar(&c.flagAutoName, "tls-auto", "",
+		"MutatingWebhookConfiguration name. If specified, will auto generate cert bundle.")
+	c.flagSet.StringVar(&c.flagAutoHosts, "tls-auto-hosts", "",
+		"Comma-separated hosts for auto-generated TLS cert. If specified, will auto generate cert bundle.")
+	c.flagSet.StringVar(&c.flagSecretName, "secret-name", "",
+		"Name of the secret to update TLS certificates")
+	c.flagSet.StringVar(&c.flagSecretNamespace, "secret-namespace", "",
+		"Namespace of the secret to update TLS certificates")
+	c.flagSet.StringVar(&c.flagCertFile, "tls-cert-file", "",
+		"PEM-encoded TLS certificate to serve. If blank, will generate random cert.")
+	c.flagSet.StringVar(&c.flagKeyFile, "tls-key-file", "",
+		"PEM-encoded TLS private key to serve. If blank, will generate random cert.")
+
+	c.httpFlags = &flags.HTTPFlags{}
+	flags.Merge(c.flagSet, c.httpFlags.Flags())
+	c.help = flags.Usage(help, c.flagSet)
+}
+
+func (c *Command) Run(args []string) int {
+	ctrl.SetLogger(zap.New(zap.UseDevMode(true)))
+	c.once.Do(c.init)
+	if err := c.flagSet.Parse(args); err != nil {
+		certGenLog.Error(err, "parsing flagSet")
+		return 1
+	}
+	if len(c.flagSet.Args()) > 0 {
+		certGenLog.Error(errors.New("should have no non-flag arguments"), "invalid arguments")
+		return 1
+	}
+
+	if c.clientset == nil {
+		config, err := rest.InClusterConfig()
+		if err != nil {
+			certGenLog.Error(err, "Error loading in-cluster K8S config")
+			return 1
+		}
+		c.clientset, err = kubernetes.NewForConfig(config)
+		if err != nil {
+			certGenLog.Error(err, "Error creating K8S client")
+			return 1
+		}
+	}
+
+	var certSource cert.Source = &cert.GenSource{
+		Name:   "Consul Webhook Certificates",
+		Hosts:  strings.Split(c.flagAutoHosts, ","),
+		Expiry: 3 * time.Minute,
```

Comment:
This will default to 24 hours (which is the default). I had this set to 3 minutes to test how the deployment behaves